Testscripts/Linux/validate-da.sh

```diff
@@ -180,53 +195,65 @@ function enable_disable_da(){
-    time_limit=$(($service_log_time + 60))
-    if [ "$DISTRO" == "debian"*] || ["$DISTRO" == "ubuntu"* ]; then
-        time_limit=$(($time_limit + 60))
-    fi
+    service_log_time=$(date -r $da_log_dir/service.log +%s)
+    time_limit=$(($service_log_time + $enable_timeout))
 
-    while [ ! -f "/var/opt/microsoft/dependency-agent/log/service.log.1" ]
+    while [ ! -f "$da_log_dir/service.log.1" ]
     do
         current_time=$(date +%s)
         if [ $current_time -gt $time_limit]; then
-            SetTestStateFailed
-            return
+            fail_test "Exceeded time limit to check service.log.1 file"
         fi
         sleep 5
     done
 
-    verify_expected_file "/var/opt/microsoft/dependency-agent/log/service.log.1"
-    [ -f "/var/opt/microsoft/dependency-agent/log/service.log.2" ] && fail_test "/var/opt/microsoft/dependency-agent/log/service.log.2 exist."
-    [ ! -c "/dev/msda" ] && fail_test "/dev/msda does not exist."
-
+    verify_expected_file "$da_log_dir/service.log.1"
+    
+    if [ -f "$da_log_dir/service.log.2" ]; then
+        fail_test "$da_log_dir/service.log.2 exist."
+    fi
+    
+    if [ ! -c "/dev/msda" ]; then 
+        fail_test "/dev/msda does not exist."
+    fi
     # Check for "driver setup status=0" and "starting the dependency agent" in service.log.1 and service.log respectively
-    ! grep -iq "driver setup status=0" /var/opt/microsoft/dependency-agent/log/service.log.1 && fail_test "driver setup status=0 not found"
-    ! grep -iq "starting the dependency agent" /var/opt/microsoft/dependency-agent/log/service.log && fail_test starting the dependency agent not found
+    if ! grep -iq "driver setup status=0" $da_log_dir/service.log.1; then
+        fail_test "driver setup status=0 not found"
+    fi
+
+    if ! grep -iq "starting the dependency agent" $da_log_dir/service.log; then
+        fail_test "starting the dependency agent not found"
+    fi
 
     # Wait for DA to be running and verify by checking for MicrosoftDependencyAgent.log
     sleep 60
-    verify_expected_file "/var/opt/microsoft/dependency-agent/log/MicrosoftDependencyAgent.log"
+    verify_expected_file "$da_log_dir/MicrosoftDependencyAgent.log"
     verify_da_pid
 
     # Wait for events and verify by checking for /var/opt/microsoft/dependency-agent/storage/*.bb files
     sleep 120
-    ! ls /var/opt/microsoft/dependency-agent/storage/*.bb && fail_test "/var/opt/microsoft/dependency-agent/storage/*.bb does not exist."
+    if ! ls $da_storage_dir/*.bb>/dev/null; then
+        fail_test "$da_storage_dir/*.bb does not exist."
+    fi
 
     # Wait for more events and verify by checking for /var/opt/microsoft/dependency-agent/storage/*.hb files
     sleep 90
-    ! ls /var/opt/microsoft/dependency-agent/storage/*.hb && fail_test "/var/opt/microsoft/dependency-agent/storage/*.hb does not exist."
+    if ! ls $da_storage_dir/*.hb>/dev/null; then
```

Comment:
Also send stderr to `/dev/null`.
Please consider adding spaces around the redirection operator for readability.